components/formats-bsd/src/loci/formats/in/OMETiffReader.java

```diff
@@ -1056,7 +1063,14 @@ public MetadataStore getMetadataStoreForConversion() {
     int realSeries = getSeries();
     for (int i=0; i<getSeriesCount(); i++) {
       setSeries(i);
-      store.setPixelsBinDataBigEndian(new Boolean(!isLittleEndian()), i, 0);
+      if (meta.getPixelsBinDataCount(i) > 0) {
+        for (int j=0; j<meta.getPixelsBinDataCount(i); j++) {
+          store.setPixelsBinDataBigEndian(new Boolean(!isLittleEndian()), i, j);
+        }
+      }
+      else {
+        store.setPixelsBigEndian(Boolean.TRUE, i);
```

Comment:
A deeper question for me would be this: will an OME-TIFF ever contain BinData elements?  If it contains TiffData, then it can't since they are mutually exclusive.  When using a metadatastore for writing, the question for me is this: who has the responsibility of cleaning up the store to match what is being written?  If we're adding BinData, that's fine, but whatever is adding it should be adding it so that the store is coherent: remove all the TiffData, and when you add the BinData, add the BinData in full, rather than a single property, so it's still valid.

Should, for example, the OME TIFF writer strip out all BinData, TiffData and MetadataOnly elements prior to writing? Should FormatWriter do this on a deep copy of the store?  Or should each reader, or FormatReader override getMetadataStore to pass a cleaned up copy?

Right now, we seem to be forcing the OmeTiffReader to partially mangle the store, but without any cleanup of TiffData.  Why this reader and not the others?  I feel that having that sanitisation in a more generic place like MetadataTools which would be called by FormatWriter would be a cleaner and more general solution to the problem.